include/roscco/ros_to_oscc.h

```diff
@@ -13,6 +13,52 @@ extern "C" {
 #include <roscco/EnableDisable.h>
 #include <roscco/SteeringCommand.h>
 #include <roscco/ThrottleCommand.h>
+#include <std_msgs/Bool.h>
+
+// struct to enable detection of a rising or falling e_stop flag.
+struct e_stop_status {
+    bool current_e_stop_message{false};
+    bool previous_e_stop_message{false};
+    bool e_stop{false};
+
+    void setPrevious() {
+        previous_e_stop_message = current_e_stop_message;
+    }
+
+    bool isRising() {
+        if((!previous_e_stop_message) && current_e_stop_message) {
+            return true;
+        }
+        else {
+            return false;
+        }
+    }
+
+    bool isFalling() {
+        if(previous_e_stop_message && (!current_e_stop_message)) {
+            return true;
+        }
+        else {
+            return false;
+        }
+    }
+
+    void checkForEvent() {
+        if (isRising()) {
+            e_stop = true;
+            ROS_DEBUG("OSCC_NODE: an emergency stop has been requested");
+        }
+        else if (isFalling()) {
+            e_stop = false;
+            ROS_DEBUG("OSCC_NODE: an emergency stop has been cancelled");
+        }
+
+        if (e_stop != current_e_stop_message) {
```

Comment:
What's the difference between `current_e_stop_message` and `e_stop`?  From my understanding, `current_e_stop_message` is being used to capture the latest e_stop command, but `current_e_stop_message` seems to be always the same:

|previous_e_stop_message | current_e_stop_message|e_stop|Description|
|:--:|:--:|:--:|:--:|
|0|0|0|this is always the same|
|0|1|1|rising|
|1|0|0|falling|
|1|1|1|this is always the same|

I might be wrong on this, but if the `current_e_stop_message` is always the same as `e_stop`, wouldn't this mean we don't need to keep track `previous_e_stop_message`?

Also, this seems to be a good thing to add to the `setCurrent` instead, the caller just needs to know how to set the current value, the internal detail will do its logic to actually set the command.